Automattic/wp-calypso
# Support chat using Zendesk Messaging

Well, this one is a doozy.

We're switching from Happy Chat to Zendesk Messaging for live chat support. This PR is big, but it's also the smallest I could get given the scope of the project. Any cleanup of Happy Chat code, old contact forms, etc. will be done as a follow-up. I want to keep this PR as easy to revert as possible if needed.

## Proposed Changes

* Disable Happy Chat feature flag and any availability checks that trigger connections to the websocket.
* Load the Zendesk Messaging script instead.
* Authenticate using already established endpoint (just with `type=zendesk`).
* Check availability using the `/help/messaging/is-available` endpoint.
* Pass the initial chat data using Zendesk user fields.
* Use the support history to figure out if the user has ongoing conversations or not (thankfully, a Messaging conversation is a "ticket"!).
* Additionally, make sure the Zendesk launcher is visible if any unread messages are reported by Zendesk's APIs directly. This is a good safeguard and makes it more dynamic.
* If the browser is blocking requests to Zendesk, show a notice and link to relevant support doc.

## Testing Instructions

The live chat support flow should feel the same up to the point where you click the chat CTA (`Chat with us`/`Still chat with us`). At that point, the Help Center should get hidden, while the Zendesk launch and widget should become visible.

<img width="558" alt="Screenshot 2023-05-16 at 19 05 25" src="https://github.com/Automattic/wp-calypso/assets/3392497/ad8af6a2-0bd5-4aa5-b3ea-3c34060b85c9">

Trying to show the Help Center should automatically minimize the Zendesk widget, to avoid clutter and confusing UI. The Zendesk launcher remains visible, so the user can come back to the conversation at any time.

If you have Enhanced Tracking Protection (or similar strict privacy settings and/or ad-blockers), you will see this notice when choosing the Live Chat support option:
<img width="432" alt="Screenshot 2023-05-16 at 19 07 32" src="https://github.com/Automattic/wp-calypso/assets/3392497/53281511-d24d-4ed0-9a5e-878acef9a7a3">

Other support options should not show this notice, of course.

Verify that the Zendesk widget follows you on different paths in Calypso. Where Help Center is, the widget should also be there.

If you have an active, ongoing conversation, it should be detected and on refreshes the launcher should appear. If new messages arrive, it will be reflected with a count/notification on the launcher as well.

<img width="265" alt="Screenshot 2023-05-16 at 17 41 35" src="https://github.com/Automattic/wp-calypso/assets/3392497/6c7a1738-88fd-4613-9fed-7624eb9d0995">

Verify these behaviors in different environments that Help Center has to work in:
 - Simple sites
 - Atomic sites
 - Editor

## Pre-merge Checklist
